src/main/java/org/folio/event/service/UserTenantService.java

```diff
@@ -75,11 +76,41 @@ public Future<String> getConsortiaCentralTenantId(Conn conn, Map<String, String>
    */
   public Future<Boolean> isConsortiaTenant(Conn conn, Map<String, String> okapiHeaders) {
     String okapiTenantId = TenantTool.tenantId(okapiHeaders);
-    Criterion criterion = new Criterion().setLimit(new Limit(1)).setOffset(new Offset(0));
-    return tenantRepository.fetchUserTenants(conn, okapiTenantId, criterion)
+    return fetchFirstUserTenant(conn, okapiTenantId)
       .map(res -> res.getTotalRecords() > 0);
   }
 
+  /**
+   * User-tenant table in each member ECS tenant has only single record, in this case http requests to this tenant allowed
+   * (central ECS tenant contains multiple user tenant associations necessary for login, saml-login, forgot password/username functionality).
+   * This method deletes  record from user-tenant table in Member ECS tenant and after this each request to that tenant will be
+   * forbidden.
+   *
+   * @param tenantId the tenant id
+   * @param vertx the vertx instance
+   * @return future with true if record was deleted or false otherwise
+   * @throws IllegalStateException if deleting is not possible
+   */
+  public Future<Boolean> deleteMemberUserTenant(String tenantId, Vertx vertx) {
+    PostgresClient pgClient = pgClientFactory.apply(vertx, tenantId);
+    return pgClient.withConn(conn -> fetchFirstUserTenant(conn, tenantId)
+      .compose(res -> {
+        if (res.getTotalRecords() > 1) {
```

Comment:
If we don't have any additional details them maybe:
```
if total_records != 0 then
  throw "should contains exactly one record";
}
```

Reply:
Updated, it makes sence for me as well to simplify, thanks